inc/namespace.php

```diff
@@ -48,7 +48,9 @@ function is_site_public() : bool {
 	}
 
 	// If there are no overrides, return whether the site is set to public.
-	return get_site()->public;
+	$site = get_site();
+
+	return $site instanceof WP_Site ? $site->public : false;
```

Comment:
I think `null` coalesce will be easier to read than this eg:
```suggestion
	return get_site()->public ?? false;
```

Reply:
It happened during install - I had an error related to the WP CLI command `wp core install...` so I ran it manually.
your suggested change still has `get_site()->public` though? does the `??` prevent the non object property access?

Reply:
@pdewouters yep indeed - the null coalescing operator checks both if the property exists and whether it's null without throwing any warnings or errors. 

You can use it in situations like this or where you might normally have had a long `isset()` check like `isset( $array[0], $array[0]['thing'], $array[0]['thing']['value'] ) ? $array[0]['thing']['value'] : 'fallback'`. You can just do `$array[0]['thing']['value'] ?? 'fallback'`